logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultAnyAssertions.kt

```diff
@@ -54,8 +55,13 @@ class DefaultAnyAssertions : AnyAssertions {
             .downCastTo(subType)
             .build()
 
-    override fun <T> isNotIn(container: AssertionContainer<T>, expected: List<T>): Assertion {
-        val assertions = expected.map { assertionBuilder.representationOnly.failing.withRepresentation(it).build() }
+    override fun <T> isNotIn(container: AssertionContainer<T>, expected: Iterable<T>): Assertion {
+        val assertions = expected.map { value ->
+            assertionBuilder.representationOnly
+                .withTest { container.maybeSubject.fold(trueProvider) { it != value } }
```

Comment:
my bad, there is a better overload than I suggested in the last comment:
```suggestion
                .withTest(container) { it != value }
```